### DISCUSSION TITLE
Unable to build the schema

### DISCUSSION THREAD

**Discussion by lrangar** [bmeares/Meerschaum#99] [Q&A] [answered]
Hi,

I have a Ubuntu instance on AWS. And I am trying to install and configure mrsm on my Ubuntu instance. The installation of msrm was successful and I was able to stack up the database using the command `mrsm stack up -d db`. However, when I am trying to build the schema using the command `mrsm cii build schema` I am getting the following error. 

```
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

(Background on this error at: https://sqlalche.me/e/14/e3q8)
```
My Ubuntu version is 
```
Distributor ID:	Ubuntu
Description:	Ubuntu 22.04.1 LTS
Release:	22.04
```
mrsm version ``` Meerschaum v1.4.13 ```.

Could someone help me fix the issue ? Thanks in advance :)

**Comment by lrangar** [comment]
FYI: I looked if there are any other processes which are using the port `5432`. But there are no processes running on the aforementioned port. My assumption is my default mrsm stack is not getting started on the port `5432` (though the output after running this command `mrsm stack up -d db` is `mrsm_db_1 is up-to-date` ). And I am clueless how to deal with this issue.

**Comment by bmeares** [comment, marked as answer]
Hi @lrangar , thanks for your question! `mrsm stack` is a wrapper for `docker-compose`, so you can try other Docker Compose commands to troubleshoot, e.g.:

```bash
mrsm stack ps
```

should print this output:

```
  Name                 Command               State                    Ports
---------------------------------------------------------------------------------------------
mrsm_db_1   docker-entrypoint.sh postg ...   Up      0.0.0.0:5433->5432/tcp,:::5433->5432/tcp
```

You can also inspect and edit the stack configuration with `mrsm edit config stack`.

Are you able to verify that the database is running and listening on port 5432? What do the logs look like for `db`?

```bash
mrsm stack logs db
```

I hope this helps! If the container is running, then the issue might be network-related. Another issue (#51) was resolved because the server did not have enough RAM to start the database, so depending on what you've allocated to your EC2, that might be the case.

Lastly, if you aren't able to get the database running on this server, you can edit the connection details for `sql:main` with `mrsm edit config` to point to an external database (or [register a new connector](https://meerschaum.io/reference/connectors/#-creating-a-connector) to use for your `cii` plugin).

**Reply by lrangar** [reply]
@bmeares Thank you for replying. Turns out that the issue is due to the insufficient RAM space as confirmed from the logs.

```
2022-12-09 23:07:11.082 UTC [1] LOG:  database system is shut down
db_1       | 
db_1       | PostgreSQL Database directory appears to contain a database; Skipping initialization
db_1       | 
db_1       | 2022-12-09 23:08:11.645 UTC [1] FATAL:  could not map anonymous shared memory: Out of memory
db_1       | 2022-12-09 23:08:11.645 UTC [1] HINT:  This error usually means that PostgreSQL's request for a shared memory segment exceeded available memory, swap space, or huge pages. To reduce the request size (currently 1166860288 bytes), reduce PostgreSQL's shared memory usage, perhaps by reducing shared_buffers or max_connections.
db_1       | 2022-12-09 23:08:11.645 UTC [1] LOG:  database system is shut down
```

Will try to increase the RAM space allocated to my instance and then continue working on building schema.

**Reply by bmeares** [reply]
You may be able to start the database with lower RAM requirements by changing the shared memory value with `mrsm edit config stack`. There are two values set to 1024 MB under `db` that would need to be changed (on mobile so I can't find the exact keys at the moment).
